src/binary_tree.rs

```diff
@@ -0,0 +1,173 @@
+//! TODO add module-level documentation
+//! TODO add more detailed documentation for all public functions/structs
+
+mod sparse_binary_tree;
+use sparse_binary_tree::NodeOrientation;
+pub use sparse_binary_tree::{Coordinate, InputLeafNode, Mergeable, Node, SparseBinaryTree};
+
+mod binary_tree_path;
+mod dapol_node;
+
+// ===========================================
+// Helper functions that need to be made available to all submodules but kept private from calling code.
+
+impl<C: Clone> Node<C> {
+    /// Returns left if this node is a left sibling and vice versa for right.
+    /// Since we are working with a binary tree we can tell if the node is a left sibling of the above layer by checking the x_coord modulus 2.
+    /// Since x_coord starts from 0 we check if the modulus is equal to 0.
+    fn node_orientation(&self) -> NodeOrientation {
+        if self.get_x_coord() % 2 == 0 {
+            NodeOrientation::Left
+        } else {
+            NodeOrientation::Right
+        }
+    }
+}
+
+// ===========================================
+// Test utils for submodules.
+
+#[cfg(test)]
+mod test_utils {
```

Comment:
It could be a good idea to have a separate `test_utils.rs` file. Then any internal unit test modules can just do something like `use crate::binary_tree::test_utils::{...}` instead of `super::super::test_utils::{...}`. You would need to add `pub use binary_tree::test_utils` in `lib.rs` but at least it would be easier to read